src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -618,3 +618,138 @@ bool Terminal::IsCursorBlinkingAllowed() const noexcept
     const auto& cursor = _buffer->GetCursor();
     return cursor.IsBlinkingAllowed();
 }
+
+// Method Description:
+// - Select the sequence between delimiters defined in Settings
+// Arguments:
+// - position: the (x,y) coordinate on the visible viewport
+void Terminal::DoubleClickSelection(const COORD position)
+{
+    // if you double click a delimiter, just select that one cell
+    COORD positionWithOffsets = _ConvertToBufferCell(position);
+    auto cellChar = _buffer->GetCellDataAt(positionWithOffsets)->Chars();
+    if (_DoubleClickDelimiterCheck(cellChar))
+    {
+        SetSelectionAnchor(position);
+        return;
+    }
+
+    // scan leftwards until delimiter is found and
+    // set selection anchor to one right of that spot
+    _ExpandDoubleClickSelection_Left(position);
+
+    // scan rightwards until delimiter is found and
+    // set endSelectionPosition to one left of that spot
+    _ExpandDoubleClickSelection_Right(position);
+}
+ 
+// Method Description:
+// - Select the entire row of the position clicked
+// Arguments:
+// - position: the (x,y) coordinate on the visible viewport
+void Terminal::TripleClickSelection(const COORD position)
+{
+    SetSelectionAnchor({ 0, position.Y });

```

Comment:
The DoubleClick calls `_ConvertToBufferCell`, but this one doesn't. Is there a reason?

Reply:
Yeah! So, the only reason we convert the position in Double Click is because we need to actually read the buffer to check for the delimiter. We don't have to care what's in the buffer for a Triple Click because the selection will span from `0` to `_buffer->GetSize().RightInclusive()` (no delimiter check necessary)

---
In reply to: [292444920](https://github.com/microsoft/terminal/pull/1197#discussion_r292444920) [](ancestors = 292444920)